soc/arm/xilinx_zynqmp/pinctrl_soc.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SOC_ARM_XLNX_ZYNQMP_SOC_PINCTRL_H_
+#define ZEPHYR_SOC_ARM_XLNX_ZYNQMP_SOC_PINCTRL_H_
+
+#include <zephyr/devicetree.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/types.h>
+
+#define MIO_DEFAULT	   0x0
+#define MIO_BASE_ADDR  0xff180000
+#define MIO_PIN_OFFSET 0x04
+
+#define MIO_L0_SEL     BIT(1)
+#define MIO_L1_SEL     BIT(2)
```

Comment:
not used?

Reply:
Yes, that's unused, but I think it should be left here. It defines the layout of the register. One less thing to check for the next person using it.

Reply:
dead code is not allowed in Zephyr

Reply:
I don't think this qualifies as dead code. A lot of device drivers and SoCs define all of their registers and fields, even if they are not used yet.

Reply:
Hi @henrikbrixandersen, @gmarull! It looks like the discussion here has been stalled for some time now.

We’ve originally added the `MIO_L1_SEL` field definition for reasons stated by @henrikbrixandersen, but we’re okay with either way - keeping this define or removing it altogether.

Let us know how we should proceed with this. Thanks!

Reply:
I'd keep it.

Reply:
Ok, it won't hurt, it's just a single definition with potential future use within pinctrl_soc.h.